wgpu-core/src/device/life.rs

```diff
@@ -211,8 +211,17 @@ impl LifetimeTracker {
         });
     }
 
-    pub(crate) fn map(&mut self, value: &Arc<Buffer>) {
-        self.mapped.push(value.clone());
+    pub(crate) fn map(&mut self, buffer: &Arc<Buffer>) -> Option<SubmissionIndex> {
+        self.mapped.push(buffer.clone());
+
+        // Warning: this duplicates what is in triage_mapped()
+        let submission = self
+            .active
+            .iter_mut()
+            .rev()
+            .find(|a| a.contains_buffer(&buffer));
```

Comment:
If we already have to do the work here I think we can remove `LifetimeTracker.triage_mapped` & `LifetimeTracker.mapped` entirely which would be a nice simplification.